benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -770,10 +663,155 @@ void GraphicsBenchmarkApp::DrawExtraInfo()
+            indicesPerDrawCall -= indicesPerDrawCall % 3;
+            for (uint32_t i = 0; i < currentDrawCallCount; i++) {
+                SphereData data                 = {};
+                data.modelMatrix                = float4x4(1.0f);
+                data.ITModelMatrix              = glm::inverse(glm::transpose(data.modelMatrix));
+                data.ambient                    = float4(0.3f);
+                data.cameraViewProjectionMatrix = mCamera.GetViewProjectionMatrix();
+                data.lightPosition              = float4(mLightPosition, 0.0f);
+                data.eyePosition                = float4(mCamera.GetEyePosition(), 0.0f);
+                mDrawCallUniformBuffers[i]->CopyFromSource(sizeof(data), &data);
+
+                frame.cmd->PushGraphicsUniformBuffer(mSphere.pipelineInterface, /* binding = */ 0, /* set = */ 0, /* bufferOffset = */ 0, mDrawCallUniformBuffers[i]);
+                frame.cmd->PushGraphicsSampledImage(mSphere.pipelineInterface, /* binding = */ 1, /* set = */ 0, mAlbedoTexture.sampledImageView);
+                frame.cmd->PushGraphicsSampler(mSphere.pipelineInterface, /* binding = */ 2, /* set = */ 0, mAlbedoTexture.sampler);
+                frame.cmd->PushGraphicsSampledImage(mSphere.pipelineInterface, /* binding = */ 3, /* set = */ 0, mNormalMapTexture.sampledImageView);
+                frame.cmd->PushGraphicsSampler(mSphere.pipelineInterface, /* binding = */ 4, /* set = */ 0, mNormalMapTexture.sampler);
+                frame.cmd->PushGraphicsSampledImage(mSphere.pipelineInterface, /* binding = */ 5, /* set = */ 0, mMetalRoughnessTexture.sampledImageView);
+                frame.cmd->PushGraphicsSampler(mSphere.pipelineInterface, /* binding = */ 6, /* set = */ 0, mMetalRoughnessTexture.sampler);
+
+                uint32_t indexCount = indicesPerDrawCall;
+                // Add the remaining indices to the last drawcall
+                if (i == currentDrawCallCount - 1) {
+                    indexCount += (currentSphereCount * mSphereIndexCount - currentDrawCallCount * indicesPerDrawCall);
+                }
+                uint32_t firstIndex = i * indicesPerDrawCall;
+                frame.cmd->DrawIndexed(indexCount, /* instanceCount = */ 1, firstIndex);
+            }
+        }
+    }
+    frame.cmd->EndRenderPass();
+}
+
+void GraphicsBenchmarkApp::RenderFullscreenQuads(grfx::SwapchainPtr swapchain, PerFrame& frame, uint32_t imageIndex)
+{
+    grfx::RenderPassPtr currentRenderPass = GetRenderpass(swapchain, imageIndex);
+
+    if (pFullscreenQuadsCount->GetValue() > 0) {
+        bool singleRenderpass = pFullscreenQuadsSingleRenderpass->GetValue();
+        frame.cmd->BindGraphicsPipeline(mFullscreenQuads.pipeline);
+        frame.cmd->BindVertexBuffers(1, &mFullscreenQuads.vertexBuffer, &mFullscreenQuads.vertexBinding.GetStride());
+
+        if (singleRenderpass) {
+            currentRenderPass = swapchain->GetRenderPass(imageIndex);
```

Comment:
Is this different from the currentRenderPass's previous value?

Reply:
Obsolete